spec/controllers/items_controller_spec.rb

```diff
@@ -0,0 +1,101 @@
+require "rails_helper"
+
+RSpec.describe ItemsController do
+  context "when user is logged in" do
+    before { sign_in_user }
+    let(:wishlist) { create(:wishlist, user: controller.current_user) }
+    let(:item) { create(:item, wishlist: wishlist) }
+
+    describe "GET #new" do
+      before { get :new, wishlist_id: wishlist.id }
+
+      it "returns 200 OK" do
+        expect(response).to have_http_status(:ok)
+      end
+
+      it "renders the 'new' template" do
+        expect(response).to render_template(:new)
+      end
+
+      it "assigns a new item as @item" do
+        expect(assigns(:item)).to be_a_new(Item)
+      end
+    end
+
+    describe "POST #create" do
+      context "with valid parameters" do
+        before { post_create(wishlist_id: wishlist.id) }
+
+        it "assigns an instance variable under @item" do
+          expect(assigns(:item)).to be_a(Item)
+        end
+
+        it "persists the new item" do
+          expect(assigns(:item)).to be_persisted
+          expect { post_create(wishlist_id: wishlist.id) }.to change(Item, :count).by(1)
+        end
+
+        it "redirects to the wishlists index page" do
+          expect(response).to redirect_to(wishlist)
+        end
+      end
+
+      context "with invalid parameters" do
+        before { post_create(name: nil, wishlist_id: wishlist.id) }
+
+        it "returns 200 OK" do
+          expect(response).to have_http_status(:ok)
+        end
+
+        it "assigns an instance variable under @item" do
+          expect(assigns(:item)).to be_a(Item)
+        end
+
+        it "doesn't persist the item" do
+          expect(assigns(:item)).not_to be_persisted
+          expect { post_create(name: nil, wishlist_id: wishlist.id) }.to_not change(Item, :count)
+        end
+
+        it "renders the 'new' template" do
+          expect(response).to render_template(:new)
+        end
+      end
+    end
+  end
+
+  context "when not logged in" do
```

Comment:
I think `specs` are up for an overall refactor. We'll keep in mind your comment, but would like to push this, as it is, right now.